Error while installing on Mac

Hi James, Team

I'm running Eclipse and trying to install Baleen, but keep getting build failure errors for the Collection Readers onwards, which means it isn't building half of the tool.

Please can you help?

Cheers,

DM


Results :

Failed tests:   testMultipleDirectories(uk.gov.dstl.baleen.collectionreaders.FolderReaderTest)
  testSubDirectoriesNonRecursive(uk.gov.dstl.baleen.collectionreaders.FolderReaderTest): expected:</[]var/folders/n2/4x13f...> but was:</[private/]var/folders/n2/4x13f...>
  testModifiedFile(uk.gov.dstl.baleen.collectionreaders.FolderReaderTest)
  testSubDirectories(uk.gov.dstl.baleen.collectionreaders.FolderReaderTest)
  testCreateFile(uk.gov.dstl.baleen.collectionreaders.FolderReaderTest)

Tests run: 19, Failures: 5, Errors: 0, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Baleen ............................................ SUCCESS [  2.281 s]
[INFO] Baleen Core ....................................... SUCCESS [ 28.455 s]
[INFO] Baleen UIMA ....................................... SUCCESS [ 10.691 s]
[INFO] Baleen Resources .................................. SUCCESS [ 44.327 s]
[INFO] Baleen Annotators ................................. SUCCESS [01:07 min]
[INFO] Baleen Collection Readers ......................... FAILURE [ 27.084 s]
[INFO] Baleen Consumers .................................. SKIPPED
[INFO] Baleen History .................................... SKIPPED
[INFO] Baleen Runner ..................................... SKIPPED
[INFO] Baleen Javadoc .................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 03:00 min
[INFO] Finished at: 2015-10-19T23:10:03+00:00
[INFO] Final Memory: 21M/208M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.10:test (default-test) on project baleen-collectionreaders: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/User1/Desktop/baleen-master/baleen/baleen-collectionreaders/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :baleen-collectionreaders

--- a/baleen/baleen-collectionreaders/src/test/java/uk/gov/dstl/baleen/collectionreaders/FolderReaderTest.java
+++ b/baleen/baleen-collectionreaders/src/test/java/uk/gov/dstl/baleen/collectionreaders/FolderReaderTest.java
@@ -20,6 +20,7 @@ import org.apache.uima.jcas.JCas;
 import org.apache.uima.jcas.tcas.DocumentAnnotation;
 import org.junit.After;
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
 
 import uk.gov.dstl.baleen.uima.BaleenCollectionReader;
@@ -31,6 +32,19 @@ public class FolderReaderTest {
 	private static final String DIR = "baleen-test";
 	File inputDir;
 	JCas jCas;
+	
+	private static Long TIMEOUT = 1000L;
+	
+	@BeforeClass
+	public static void beforeClass(){
+		//If we're testing on a Mac, then we need to set the time out higher,
+		//as currently the WatchService on a Mac uses polling rather than a
+		//native implementation and therefore we need to ensure we wait longer
+		//than the poll interval
+		if(System.getProperty("os.name").toLowerCase().startsWith("mac os x")){
+			TIMEOUT = 15000L;
+		}
+	}
 
 	@Before
 	public void beforeTest() throws Exception{
@@ -73,13 +87,13 @@ public class FolderReaderTest {
 		f.createNewFile();
 
 		//Wait for file to be written and change detected
-		Thread.sleep(1000);
+		Thread.sleep(TIMEOUT);
 
 		assertTrue(bcr.doHasNext());
 
 		bcr.getNext(jCas.getCas());
 
-		assertEquals(f.getPath(), getSource(jCas));
+		assertFilesEquals(f.getPath(), getSource(jCas));
 
 		assertFalse(bcr.doHasNext());
 
@@ -104,7 +118,7 @@ public class FolderReaderTest {
 		File f22 = new File(inputDir2, TEST2_FILE);
 		f22.createNewFile();
 
-		Thread.sleep(1000);
+		Thread.sleep(TIMEOUT);
 
 		assertNextSourceNotNull(bcr);
 		assertNextSourceNotNull(bcr);
@@ -120,7 +134,7 @@ public class FolderReaderTest {
 	public void testSubDirectories() throws Exception{
 		File subdir = new File(inputDir, "subdir");
 		subdir.mkdir();
-
+		
 		File f1 = new File(subdir, TEXT1_FILE);
 		f1.createNewFile();
 
@@ -131,7 +145,7 @@ public class FolderReaderTest {
 		File f2 = new File(subdir, TEST2_FILE);
 		f2.createNewFile();
 
-		Thread.sleep(1000);
+		Thread.sleep(TIMEOUT);
 
 		assertNextSourceNotNull(bcr);
 
@@ -157,18 +171,18 @@ public class FolderReaderTest {
 
 		assertTrue(bcr.hasNext());
 		bcr.getNext(jCas.getCas());
-		assertEquals(f2.getPath(), getSource(jCas));
+		assertFilesEquals(f2.getPath(), getSource(jCas));
 		
 		jCas.reset();
 		
 		File f3 = new File(inputDir, TEST3_FILE);
 		f3.createNewFile();
 		
-		Thread.sleep(1000);
+		Thread.sleep(TIMEOUT);
 		
 		assertTrue(bcr.hasNext());
 		bcr.getNext(jCas.getCas());
-		assertEquals(f3.getPath(), getSource(jCas));
+		assertFilesEquals(f3.getPath(), getSource(jCas));
 		
 		bcr.close();
 
@@ -187,12 +201,12 @@ public class FolderReaderTest {
 		f.createNewFile();
 
 		//Wait for file to be written and change detected
-		Thread.sleep(1000);
+		Thread.sleep(TIMEOUT);
 
 		assertTrue(bcr.doHasNext());
 
 		bcr.getNext(jCas.getCas());
-		assertEquals(f.getPath(), getSource(jCas));
+		assertFilesEquals(f.getPath(), getSource(jCas));
 
 		jCas.reset();
 
@@ -201,13 +215,13 @@ public class FolderReaderTest {
 		writer.write("Test");
 		writer.close();
 
-		Thread.sleep(1000);
+		Thread.sleep(TIMEOUT);
 
 		assertTrue(bcr.doHasNext());
 
 		bcr.getNext(jCas.getCas());
 
-		assertEquals(f.getPath(), getSource(jCas));
+		assertFilesEquals(f.getPath(), getSource(jCas));
 		assertEquals("Test", jCas.getDocumentText().trim());
 
 		assertFalse(bcr.doHasNext());
@@ -225,12 +239,12 @@ public class FolderReaderTest {
 		f.createNewFile();
 
 		//Wait for file to be written and change detected
-		Thread.sleep(1000);
+		Thread.sleep(TIMEOUT);
 
 		f.delete();
 
 		//Wait for file to be written and change detected
-		Thread.sleep(1000);
+		Thread.sleep(TIMEOUT);
 
 		assertFalse(bcr.doHasNext());
 
@@ -281,5 +295,12 @@ public class FolderReaderTest {
 	private String getSource(JCas jCas){
 		DocumentAnnotation doc = (DocumentAnnotation) jCas.getDocumentAnnotationFs();
 		return doc.getSourceUri();
+	}
+	
+	private void assertFilesEquals(String s1, String s2) throws IOException{
+        File f1 = new File(s1);
+        File f2 = new File(s2);
+
+        assertTrue(Files.isSameFile(f1.toPath(), f2.toPath()));
 	}
 }

